### PR TITLE
Apply eslint-rule guard-for-in - Related to #1597

### PR DESCRIPTION
**Pull request by diego-G**
### Review checklist

* The PR is related to #1597
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: modules/blocks/verify.js [hunk 1]
```diff
@@ -319,27 +319,29 @@ __private.verifyPayload = function(block, result) {
...
 	for (const i in block.transactions) {
-		const transaction = block.transactions[i];
-		let bytes;
-
-		try {
-			bytes = library.logic.transaction.getBytes(transaction);
-		} catch (e) {
-			result.errors.push(e.toString());
-		}
+		if (Object.prototype.hasOwnProperty.call(block.transactions, i)) {
```
Comment (MaciejBaj) [inline modules/blocks/verify.js]: `block.transactions` is an array, I don't think this check is necessary here and introduces confusion in the critical function - `blocks.verify`.

Comment (MaciejBaj) [reply modules/blocks/verify.js]: Overall this rule looks odd to me. We should either introduce a rule of not-using `for ... in` at all, or keep this flag off.

Comment (diego-G) [reply modules/blocks/verify.js]: This code will change when introducing `no-restricted-syntax` https://github.com/LiskHQ/lisk/pull/2643 . Perhaps it'll make more sense after it.

**Comment by MaciejBaj** [comment]
Please solve the conflicts:
```
modules/blocks/verify.js
modules/loader.js
modules/node.js
```

**Comment by diego-G** [comment]
Rule doesn't trigger any error. It will be removed on the PR #2649 instead.

**Pull request closed without merging**